File: objective-c/Orientation/v1/Orientation/ViewControllers/RootNavigationController.m
//
//  RootNavigationController.m
//  Orientation
//
//  Created by Alexandr Kirilov on 15/04/2020.
//  Copyright © 2020 Alexandr Kirilov. All rights reserved.
//

#import "RootNavigationController.h"

@interface RootNavigationController ()

@end

@implementation RootNavigationController

-(void) loadView {
	
	[super loadView];
	DemoViewController* oDemoVC = [[DemoViewController alloc] init];
	[self setViewControllers:@[oDemoVC]];
}

-(void) viewDidLoad {
	
    [super viewDidLoad];
}

@end

File: swift/UIKit/UIKit UISearchController/UIKit UISearchController/Primitives/RootNavigationController.swift
//
//  RootNavigationController.swift
//  UIKit UISearchController
//
//  Created by Alexandr Kirilov on 30/03/2020.
//  Copyright © 2020 Alexandr Kirilov. All rights reserved.
//

import UIKit

class RootNavigationController: UINavigationController {
	
	let pNavBarBackgroundColor: UIColor = __COLOR_BLUE_DARK;
	let pNavBarTextColor: UIColor = __COLOR_WHITE;
	
	override var preferredStatusBarStyle: UIStatusBarStyle {

		return .lightContent;
	}

    override func viewDidLoad() {
		
		super.viewDidLoad();
		
		navigationBar.barTintColor = pNavBarBackgroundColor;
		navigationBar.tintColor = pNavBarTextColor;
		navigationBar.titleTextAttributes = [.foregroundColor: pNavBarTextColor];
		navigationBar.isTranslucent = false;
//		navigationBar.barStyle = .black;
		
		if #available(iOS 11.0, *) {
			navigationBar.prefersLargeTitles = true;
			navigationItem.largeTitleDisplayMode = .always;
			view.backgroundColor = pNavBarBackgroundColor;
			navigationBar.largeTitleTextAttributes = [NSAttributedString.Key.foregroundColor: pNavBarTextColor];
		}
    }
}
